openpype/modules/slack/plugins/publish/integrate_slack_api.py

```diff
@@ -0,0 +1,83 @@
+try:
+    from slackclient import SlackClient
+    python2 = True
+except ImportError:
+    python2 = False
+    from slack_sdk import WebClient
+    from slack_sdk.errors import SlackApiError
+
+import pyblish.api
+from openpype.lib.plugin_tools import prepare_template_data
+
+
+class IntegrateSlackAPI(pyblish.api.InstancePlugin):
+    """ Send message notification to a channel.
+
+        Triggers on instances with "slack" family, filled by
+        'collect_slack_family'.
+
+        Expects configured profile in
+        Project settings > Slack > Publish plugins > Notification to Slack
+
+        Message template can contain {} placeholders from anatomyData.
+    """
+    order = pyblish.api.IntegratorOrder + 0.499
+    label = "Integrate Slack Api"
+    families = ["slack"]
+
+    optional = True
+
+    def process(self, instance):
+        message_templ = instance.data["slack_message"]
+
+        fill_pairs = set()
+        for key, value in instance.data["anatomyData"].items():
+            if not isinstance(value, str):
+                continue
+            fill_pairs.add((key, value))
+
+        message = message_templ.format(**prepare_template_data(fill_pairs))
+
+        self.log.debug("message:: {}".format(message))
+        if '{' in message:
```

Comment:
I wouldn't skip message with `{` in message (it's valid symbol). And it would crash on line 39 if there are missing data :)

Reply:
Changed it to try except